Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -691,6 +692,20 @@ class SubprocessFastWatcherTests(SubprocessWatcherMixin,
 
         Watcher = unix_events.FastChildWatcher
 
+    def has_pidfd_support():
+        if not hasattr(os, 'pidfd_open'):
+            return False
+        try:
+            os.close(os.pidfd_open(os.getpid()))
+        except OSError:
+            return False
+        return True
+
+    if has_pidfd_support():
+        class SubprocessPidfdWatcherTests(SubprocessWatcherMixin,
```

Comment:
Good idea w/ `os.close(os.pidfd_open(os.getpid()))`, I hadn't thought of that. Minor recommendation:
```python
@unittest.skipUnless(has_pidfd_support(), "OS does not support pidfd_open")
class SubprocessPidfdWatcherTests(SubprocessWatcherMixin,
...
```